src/main/java/edu/ucla/library/iiif/fester/handlers/CheckEndpointsHandler.java

```diff
@@ -0,0 +1,116 @@
+     * @param aConfig A JSON configuration
+     */
+    public CheckEndpointsHandler( final Vertx aVertx, final JsonObject aConfig ) {
+        super( aVertx, aConfig );
+    }
+
+    @Override
+    public void handle( final RoutingContext aContext ) {
+        final HttpServerResponse response = aContext.response();
+        final Manifest upload = new Manifest("id","label");
+        final JsonObject endpoints = new JsonObject();
+        final JsonObject status = new JsonObject();
+
+        try {
+            final Future<Object> composite = Future.future( put -> {
+                myS3Client.put(myS3Bucket, UPLOAD_KEY, Buffer.buffer(upload.toJSON().encodePrettily()), putResponse -> {
+                    final int statusCode = putResponse.statusCode();
+                    endpoints.put(Status.PUT_RESPONSE, statusCode);
+                    determineEndpointStatus(statusCode,endpoints,put);
+                });
+            } ).compose( v2 -> {
+                return Future.future( get -> {
+                    myS3Client.get(myS3Bucket, UPLOAD_KEY, getResponse -> {
+                        final int statusCode = getResponse.statusCode();
+                        endpoints.put(Status.GET_RESPONSE, statusCode);
+                        determineEndpointStatus(statusCode,endpoints,get);
+                    });
+                } );
+            } ).compose( v3 -> {
+                return Future.future( delete -> {
+                    myS3Client.delete(myS3Bucket, UPLOAD_KEY, deleteResponse -> {
+                        final int statusCode = deleteResponse.statusCode();
+                        endpoints.put(Status.DELETE_RESPONSE, statusCode);
+                        determineEndpointStatus(statusCode,endpoints,delete);
+                    });
+                } );
+            } );
+
+            composite.onSuccess( success -> {
+                status.put( Status.STATUS, determineOverallStatus(endpoints) ).put( Status.ENDPOINTS, endpoints );
+
+                response.setStatusCode(200);
+                response.putHeader(Constants.CONTENT_TYPE, Constants.JSON_MEDIA_TYPE).end(status.encodePrettily());
+            } ).onFailure( failure -> {
+                response.setStatusCode(HTTP.INTERNAL_SERVER_ERROR);
+                response.putHeader(Constants.CONTENT_TYPE, Constants.PLAIN_TEXT_TYPE);
+                response.end("failed somehow\n");
+            } );
+
+        } catch (final Throwable aThrowable) {
+            final String exceptionMessage = aThrowable.getMessage();
+
+            LOGGER.error(aThrowable, exceptionMessage);
+
+            response.setStatusCode(HTTP.INTERNAL_SERVER_ERROR);
+            response.putHeader(Constants.CONTENT_TYPE, Constants.PLAIN_TEXT_TYPE);
+            response.end(exceptionMessage);
+        }
+
+    }
+
+    private void determineEndpointStatus(final int aCode, final JsonObject aMessage, final Promise<Object> aPromise) {
+        if (aCode >= 400 && aCode < 500 ) {

```

Comment:
Switched to HTTP constants instead of literals; FWIW, HTTP.java does not include all 4XX/5XX codes--should I add those now, or do as another ticket in next sprint?